std/haxe/io/Path.hx

```diff
@@ -230,8 +230,13 @@ class Path {
 		var acc = new StringBuf();
 		var colon = false;
 		var slashes = false;
+		#if hl
```

Comment:
should it not be #if utf16 ?

Reply:
It was (in bba3863), but that broke the Java tests.

Reply:
It would be interesting to know why this break the Java tests... And what about JS ? it's also utf16.

Reply:
#8239 fixed this. I changed this back to `#if utf16` just now. JS is actually works with both versions.